From 22498811fcdaf369a41e178262c61485f4e3fb5c Mon Sep 17 00:00:00 2001
From: Chase Douglas <chase.douglas@canonical.com>
Date: Tue, 14 Feb 2012 14:46:07 -0800
Subject: [PATCH 06/10] Disable scrolling when beginning a clickpad press

There really isn't a point to scrolling while a clickpad is pressed. In
particular, the clickpad button areas and the horizontal edge scrolling
areas overlap, so horizontal edge scrolling must be disabled. Also,
performing two finger scrolling while a third finger presses the button
would require us to inhibit touch events until four touches are present.
That is enough reason to disable two finger scrolling as well.

Signed-off-by: Chase Douglas <chase.douglas@canonical.com>
---
 src/synaptics.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/synaptics.c b/src/synaptics.c
index 77f9f2d..f9e591e 100644
--- a/src/synaptics.c
+++ b/src/synaptics.c
@@ -1850,6 +1850,12 @@ HandleTapProcessing(SynapticsPrivate *priv, struct SynapticsHwState *hw,
 	}
 	break;
     case TS_CLICKPAD_MOVE:
+	/* Disable scrolling once a button is pressed on a clickpad */
+	priv->vert_scroll_edge_on = FALSE;
+	priv->horiz_scroll_edge_on = FALSE;
+	priv->vert_scroll_twofinger_on = FALSE;
+	priv->horiz_scroll_twofinger_on = FALSE;
+
         /* Assume one touch is only for holding the clickpad button down */
 	if (hw->numFingers > 1)
 	    hw->numFingers--;
-- 
1.7.9.2

